### DISCUSSION TITLE
Bug in setDataFromResource for Areablock?

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#16672] [Q&A / Support]
Hi,

in a older Version there is a if statement for passing an array or a string. This causes an error in the current version when calling Tool\Serialize::unserialize($data);

Is it a Bug or should it be 

```
if (!empty($data) && is_string($data)) {
            $unserializedData = Tool\Serialize::unserialize($data);
        } else {
            $unserializedData = $data;
        }
```
in https://github.com/pimcore/pimcore/blob/11.x/models/Document/Editable/Areablock.php as in version before?

**Comment by blankse** [comment]
@langert I created a PR: https://github.com/pimcore/pimcore/pull/16701
